hdc/algo/ops/stats.py

```diff
@@ -0,0 +1,423 @@
+                y[ri, ci, :] = -9999
+                continue
+
+            spi = np.full(t, p_zero, dtype=nt.float64)  # type: ignore
+
+            for tix in valid_ix:
+                spi[tix] = p_zero + (
+                    (1 - p_zero)
+                    * sc.gammainc(alpha, xt[tix] / beta)  # pylint: disable=no-member
+                )
+
+            for tix in range(t):
+                spi[tix] = sc.ndtri(spi[tix]) * 1000
+
+            np.round_(spi, 0, spi)
+
+            y[ri, ci, :] = spi[:]
+
+    return y
+
+
+@numba.njit
+def mk_score(x):
+    """Calculate MK score (S) and Kendall's Tau.
+
+    https://vsp.pnnl.gov/help/vsample/design_trend_mann_kendall.htm
+
+    Hussain et al., (2019). pyMannKendall:
+    a python package for non parametric Mann Kendall family of trend tests..
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    """
+    n = len(x)
+    _s1 = 0
+    _s2 = 0
+
+    for k in range(0, n - 1):
+        for kk in range(k + 1, n):
+            if x[kk] > x[k]:
+                _s1 += 1
+            if x[kk] < x[k]:
+                _s2 += 1
+
+    s = _s1 - _s2
+    tau = s / (0.5 * n * (n - 1))
+    return s, tau
+
+
+@numba.njit
+def mk_variance_s(x):
+    """Mann-Kendall's variance S.
+
+    https://vsp.pnnl.gov/help/vsample/design_trend_mann_kendall.htm
+
+    Hussain et al., (2019). pyMannKendall:
+    a python package for non parametric Mann Kendall family of trend tests..
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    """
+    xu = np.unique(x)
+    n = len(x)
+
+    if len(xu) == n:
+        return (n * (n - 1) * (2 * n + 5)) / 18
+
+    tp = 0
+    for i in range(len(xu)):  # pylint: disable=consider-using-enumerate
+        _tp = 0
+        for ii in range(n):
+            if xu[i] == x[ii]:
+                _tp += 1
+
+        tp += _tp * (_tp - 1) * (2 * _tp + 5)
+
+    return ((n * (n - 1) * (2 * n + 5)) - tp) / 18
+
+
+@numba.njit
+def mk_z_score(s, vs):
+    """Calculate Mann-Kendall's Z (MKZ).
+
+    https://vsp.pnnl.gov/help/vsample/design_trend_mann_kendall.htm
+
+    Hussain et al., (2019). pyMannKendall:
+    a python package for non parametric Mann Kendall family of trend tests..
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    """
+    if s > 0:
+        return (s - 1) / sqrt(vs)
+
+    if s < 0:
+        return (s + 1) / sqrt(vs)
+
+    return 0
+
+
+@numba.njit
+def mk_p_value(z, alpha=0.05):
+    """Calculate Mann-Kendall's p value and significance.
+
+    https://vsp.pnnl.gov/help/vsample/design_trend_mann_kendall.htm
+
+    Hussain et al., (2019). pyMannKendall:
+    a python package for non parametric Mann Kendall family of trend tests..
+    Journal of Open Source Software, 4(39), 1556, https://doi.org/10.21105/joss.01556
+    """
+    p = 2 * (1 - (0.5 * (1.0 + erf(abs(z) * sqrt(0.5)))))
+    h = int(abs(z) > sc.ndtri(1 - alpha / 2))
```

Comment:
Do you have a link to scipy code or from where you got the formula for the `cdf` and `ppf` here? That's super nice!

Reply:
see 

https://stats.stackexchange.com/questions/187828/how-are-the-error-function-and-standard-normal-distribution-function-related 

and 

https://stackoverflow.com/questions/20626994/how-to-calculate-the-inverse-of-the-normal-cumulative-distribution-function-in-p